grails-app/migrations/views/stock-movement.sql

```diff
@@ -0,0 +1,131 @@
+CREATE OR REPLACE VIEW stock_movement AS
+    select
+           r.id,
+           r.name,
+           r.request_number as identifier,
+           r.description,
+           r.origin_id,
+           r.destination_id,
+           r.date_requested,
+           r.date_created,
+           r.last_updated,
+           r.requested_by_id,
+           r.created_by_id,
+           r.updated_by_id,
+           ri.item_count,
+           'OUTBOUND' as stock_movement_type,
+           s.shipment_type_id,
+           s.expected_shipping_date as date_shipped,
+           s.expected_delivery_date,
+           s.driver_name,
+           s.additional_information as comments,
+           s.id as shipment_id,
+           s.current_status as shipment_status,
+           rn.identifier as tracking_number,
+           r.id as requisition_id,
+           r.status as requisition_status,
+           r.requisition_template_id as stocklist_id,
+           null as order_id,
+           null as order_status
+    from requisition r
+        left join shipment s on s.requisition_id = r.id
+        left join (
+            select
+                   requisition_id,
+                   count(*) as item_count
+            from requisition_item
+            group by requisition_id
+            ) ri on ri.requisition_id = r.id
+
+        left join shipment_reference_number sr on s.id = sr.shipment_reference_numbers_id
+        left join reference_number rn on rn.id = sr.reference_number_id and rn.reference_number_type_id = "10"
+    where r.is_template is false
+
+union all
```

Comment:
does this need to be union all?

Reply:
union is more expensive, because is doing ordering and removing duplicates